src/labeler.py

```diff
@@ -1,15 +1,97 @@
 
+# Labeling functions go here
+import json
+
+
 # Returns a list of map labels
 # Labels: 0 = no map in solution, >0 = map exists in solution
 # Each index in all_labels labels the corresponding index in all_problems
 def maps(all_problems):
     all_labels = []
-    label = 0
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'map<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def lists(all_problems):
+    all_labels = []
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'list<' in line.lower() or 'vector<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def sets(all_problems):
+    all_labels = []
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'set<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def stacks(all_problems):
+    all_labels = []
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'stack<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def queues(all_problems):
+    all_labels = []
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'queue<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def priority_queues(all_problems):
+    all_labels = []
+
+    for problem in all_problems:
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'priority_queue<' in line.lower():
+                    label = 1
+        all_labels.append(label)
+    return all_labels
+
+
+def linked_lists(all_problems):
+    all_labels = []
+
     for problem in all_problems:
-        for solution in problem.solutions:
-            for line in solution:
-                if 'map<' in line:
-                    label += 1
+        label = 0
+        for solution in problem['solutions']:
+            for line in solution.split('\n'):
+                if 'Node' in line.lower() or 'LinkedList<' in line.lower():
```

Comment:
There could be a node in any linked data structure like a tree, not just a linked list. Also, linked lists in C++ are just called `List`, and searching for them would also hit java's Array**List**.

Basically, I suggest removing this function and combining both linked and array lists as a single feature (at least for now) unless you can think of a clever way of differentiating them. There is an existing labeling function above called `lists()` that does this.

Also just a minor fix for the future: should be lowercase `'linkedlist<'` because there are no uppercase letters in `line.lower()`

Reply:
I added a vector labeler earlier but saw that you combined the list and vector into one function so I removed it. I wasn't sure how to incorporate the linked lists, but that makes sense! Thank you for reviewing!